### PR TITLE
Initial push of MyRocks Status variables docs and changes (5.7)

### PR DESCRIPTION
**Pull request by hrvojem**
since the experimental release.

Fixed: DOC-631, DOC-607, DOC-620, DOC-615, DOC-611, DOC-617, DOC-637,
DOC-616, DOC-646

### CODE REVIEW

**Review by george-lorch — APPROVED** [review]
Excellent!

File: doc/source/myrocks/variables.rst [hunk 1]
```diff
@@ -628,6 +663,17 @@ until there is less than 10 bits of free space remaining.
...
+.. variable:: rocksdb_bulk_load_allow_unsorted
+
+  :version 5.7.20-18: Implemented
+  :cli: ``--rocksdb-bulk-load-allow-unsorted``
+  :dyn: Yes
+  :scope: Global, Session
+  :vartype: Boolean
+  :default: ``OFF``
+
+When enabled input is sorted by Primary Key during bulkload.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: The variable name "allow unsorted" seems to be the opposite of the description "sort"?

File: doc/source/myrocks/variables.rst [hunk 2]
```diff
@@ -1977,3 +2120,610 @@ which can be useful for bulk loading.
...
+
+This variable shows the number of ``Get()`` queries served by L1.
+
+.. variable:: rocksdb_get_hit_l2_and_up
+
+This variable shows the number of ``Get()`` queries served by L2 and up.
+
+.. variable:: rocksdb_getupdatessince_calls
+
+This variable shows the number of calls to ``GetUpadtesSince`` function.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: s/Upadtes/Updates

File: doc/source/myrocks/variables.rst [hunk 3]
```diff
@@ -1977,3 +2120,610 @@ which can be useful for bulk loading.
...
+
+.. variable:: rocksdb_no_file_closes
+
+This variable shows the number of time file were closed.
+
+.. variable:: rocksdb_no_file_errors
+
+.. variable:: rocksdb_no_file_opens
+
+This variable shows the number of time file were closed.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: s/closed/opened

File: doc/source/myrocks/variables.rst [hunk 4]
```diff
@@ -1977,3 +2120,610 @@ which can be useful for bulk loading.
...
+
+This variable shows the number of currently open iterators.
+
+.. variable:: rocksdb_number_block_not_compressed
+
+This variable shows the number of uncompressed blocks.
+
+.. variable:: rocksdb_number_db_next
+
+This variable shows the number of calls to next.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: quote "next"

File: doc/source/myrocks/variables.rst [hunk 5]
```diff
@@ -1977,3 +2120,610 @@ which can be useful for bulk loading.
...
+
+.. variable:: rocksdb_stall_memtable_limit_slowdowns
+
+.. variable:: rocksdb_stall_total_stops
+
+.. variable:: rocksdb_stall_total_slowdowns
+
+.. variable:: rocksdb_stall_micros
+
+This variable shows how long (in microseconds) the writer has to wait for
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: s/has/had

File: doc/source/release-notes/Percona-Server-5.7.20-18.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+ * |Percona Server| packages are now available for *Ubuntu 17.10 (Artful)*.
+
+Bugs Fixed
+==========
+
+
+MyRocks Changes
+===============
+
+* RocksDB has implemed a FlushWAL API which improves upon the performance of
```
Comment (laurynas-biveinis) [inline doc/source/release-notes/Percona-Server-5.7.20-18.rst]: implemented

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: doc/source/myrocks/variables.rst (hunk 1 above)
Comment (george-lorch) [reply doc/source/myrocks/variables.rst]: Considering the origination of this and the feature, this is correct. Previously, bulk load would only succeed if keys/rows were added in perfect sort order. Enabling this allows rows to be added in unsorted order but at a performance penalty. It's not ideal, but it _is_ correct in MyRocks/RocksDB semantics.

Comment (laurynas-biveinis) [reply doc/source/myrocks/variables.rst]: I see, then a clarification is needed with what you wrote: when enabled, unsorted inputs are allowed, which are then sorted by the bulkloader itself, at a performance penalty

File: doc/source/myrocks/variables.rst [hunk 7]
```diff
@@ -33,9 +33,9 @@ the default value is used.
 
 Also, all variables can exist in one or both of the following scopes:
 
-* *Global* scope defines how the variable affects overall server operation.
+* A *Global* scope defines how the variable affects overall server operation.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: s/GLobaL/global,
s/Session/session below

File: doc/source/myrocks/variables.rst [hunk 8]
```diff
@@ -628,6 +663,18 @@ until there is less than 10 bits of free space remaining.
...
+.. variable:: rocksdb_bulk_load_allow_unsorted
+
+  :version 5.7.20-18: Implemented
+  :cli: ``--rocksdb-bulk-load-allow-unsorted``
+  :dyn: Yes
+  :scope: Global, Session
+  :vartype: Boolean
+  :default: ``OFF``
+
+When enabled, unsorted inputs are allowed, which are then sorted by
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: By default, the bulk loader requires its input to be sorted in the primary key order. If enabled, unsorted inputs are allowed too, which are ...

File: doc/source/myrocks/variables.rst [hunk 9]
```diff
@@ -1284,8 +1382,28 @@ Allowed range is up to ``64``.
...
+.. variable:: rocksdb_max_background_jobs
+
+  :version 5.7.20-18: Implemented
+  :cli: ``--rocksdb-max-background-jobs``
+  :dyn: Yes
+  :scope: Global
+  :vartype: Numeric
+  :default: ``2``
+
+This variable has been introduce in |Percona Server| :rn:`5.7.20-18`
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: s/introduce/introduced

File: doc/source/myrocks/variables.rst [hunk 10]
```diff
@@ -1977,3 +2121,610 @@ which can be useful for bulk loading.
...
+
+.. variable:: rocksdb_queries_range
+
+.. variable:: rocksdb_covered_secondary_key_lookups
+
+.. variable:: rocksdb_block_cache_add
+
+This variable shows the number of blocks added to block cache.
+
+.. variable:: rocksdb_block_cache_add
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: Duplicated

File: doc/source/myrocks/variables.rst [hunk 11]
```diff
@@ -1977,3 +2121,610 @@ which can be useful for bulk loading.
...
+from block cache.
+
+.. variable:: rocksdb_block_cache_filter_miss
+
+This variable shows the number of times cache miss when accessing filter
+block from block cache.
+
+.. variable:: rocksdb_block_cache_hit
+
+This variable shows the number of blocks added to block cache.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: Cache hits not number of blocks added to cache?

File: doc/source/myrocks/variables.rst [hunk 12]
```diff
@@ -1977,3 +2121,610 @@ which can be useful for bulk loading.
...
+
+This variable shows the number of bytes of index block erased from cache.
+
+.. variable:: rocksdb_block_cache_index_bytes_insert
+
+This variable shows the number of bytes of index blocks inserted into cache.
+
+.. variable:: rocksdb_block_cache_index_hit
+
+This variable shows the number of index blocks added to block cache.
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst]: Likewise

File: doc/source/myrocks/variables.rst [hunk 13]
```diff
@@ -1977,3 +2121,610 @@ which can be useful for bulk loading.
...
+.. variable:: rocksdb_block_cache_index_miss
+
+This variable shows the number of times cache hit when accessing index
+block from block cache.
+
+.. variable:: rocksdb_block_cache_miss
+
+This variable shows the total number of block cache misses.
+
+.. variable:: rocksdb_block_cachecompressed_hit
```
Comment (laurynas-biveinis) [inline doc/source/myrocks/variables.rst:2495]: @georgelorchpercona how come a missing underscore here? Possible to fix? Or is "cachecompressed" a thing, like "GetUpdatesSince" below?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: doc/source/myrocks/variables.rst (hunk 13 above)
Comment (george-lorch) [reply doc/source/myrocks/variables.rst:2495]: That's how it is defined in the code.

Comment (laurynas-biveinis) [reply doc/source/myrocks/variables.rst:2495]: Yes, my question is whether it's possible to fix it in the code?

Comment (george-lorch) [reply doc/source/myrocks/variables.rst:2495]: This seems to be in the MyRocks translation from RocksDB, not in RocksDB so yes, it looks possible to fix by just changing the name in the translation macro.

Comment (george-lorch) [reply doc/source/myrocks/variables.rst:2495]: https://jira.percona.com/browse/MYR-212

Comment (george-lorch) [reply doc/source/myrocks/variables.rst:2495]: https://github.com/percona/percona-server/pull/2003

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Comment by peterschwaller** [comment]
Dummy comment to trigger sync to Jira.  Please ignore.

**Pull request merged**